src/engine/strat_engine/backstore/metadata.rs

```diff
@@ -69,7 +114,10 @@ impl BDA {
         f.write_all(&zeroed[..SECTOR_SIZE * 6])?; // Zero 6 unused sectors
         f.sync_all()?;
 
-        let regions = mda::MDARegions::initialize(BDA_STATIC_HDR_SIZE, header.mda_size, f)?;
+        drop(f);
```

Comment:
@agrover The code between the `borrow_mut` and the associated `drop` has many places where code execution could return an error, thus not call `drop`.  So if we have an error in between here, `f` would still be borrowed mut and then we would end up in the error leg that calls initialize where we then call `r_state.restore();` which would cause a run-time error as it tries to do `rc_f.borrow_mut(); or perhaps I'm missing something here?

Reply:
@agrover If what I am outlining here is indeed possible, then utilizing an anonymous function that @mulkieran suggests frequently would be a viable solution.

Reply:
We're manually dropping the ref here b/c MDARegions::initialize will borrow it (and re-borrowing is an error) instead of waiting until the end of scope for it to be dropped automatically, but that doesn't affect the places where f will be automatically dropped, such as the error cases you mention. Alternatively we could add another scope around the whole first part of the function that would accomplish the same thing, but I thought this was preferable.

Reply:
I should probably go ahead and add a comment there.